src/App.tsx

```diff
@@ -14,11 +16,119 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [filteredGoods, setFilteredGoods] = useState(goodsFromServer);
+  const [listIsVisible, setListIsVisible] = useState(false);
+  const [goods, setGoods] = useState(filteredGoods);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverseGoods = () => {
+    setIsReversed(!isReversed);
+    setGoods(current => [...current].reverse());
+  };
+
+  const filterGoods = (value: number) => {
+    setFilteredGoods([...goodsFromServer].filter(good => good.length >= value));
+  };
+
+  return (
+    <div className="App">
+      <h1>Goods</h1>
+      {console.log(isReversed)}
+      {listIsVisible && (
+        <>
+          <GoodsList goods={goods} />
+
+          <button
+            type="button"
+            onClick={() => reverseGoods()}
```

Comment:
```suggestion
            onClick={reverseGoods}
```
you can remove the arrow function because method doesn't accept any arguments